srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -0,0 +1,260 @@
+                }
+            }
+            throw new RuntimeException(
+                    String.format("Could not locate [%s] starting at path [%s]", relativeMvnSrcdepsYaml, baseDir));
+        }
+
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(SrcdepsLocalRepositoryManager.class);
+    public static final Path relativeMvnSrcdepsYaml = Paths.get(".mvn", "srcdeps.yaml");
+
+    private final BuildService buildService;
+    private final Path scrdepsDir;
+    private final LocalRepositoryManager delegate;
+
+    private LazyConfiguration configuration;
+
+    public SrcdepsLocalRepositoryManager(LocalRepositoryManager delegate, Provider<MavenSession> sessionProvider,
+            BuildService buildService) {
+        super();
+        this.delegate = delegate;
+        this.buildService = buildService;
+        this.scrdepsDir = delegate.getRepository().getBasedir().toPath().getParent().resolve("srcdeps");
+        this.configuration = new LazyConfiguration(sessionProvider);
+    }
+
+    /**
+     * Delegated to {@link #delegate}
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#add(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalArtifactRegistration)
+     */
+    @Override
+    public void add(RepositorySystemSession session, LocalArtifactRegistration request) {
+        delegate.add(session, request);
+    }
+
+    /**
+     * Delegated to {@link #delegate}
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#add(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalMetadataRegistration)
+     */
+    @Override
+    public void add(RepositorySystemSession session, LocalMetadataRegistration request) {
+        delegate.add(session, request);
+    }
+
+    /**
+     * In case the {@link #delegate} does not find the given artifact and the given artifact's version string is a
+     * srcdeps version string, then the version is built from source and returned.
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#find(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalArtifactRequest)
+     */
+    @Override
+    public LocalArtifactResult find(RepositorySystemSession session, LocalArtifactRequest request) {
+        final LocalArtifactResult result = delegate.find(session, request);
+
+        Artifact artifact = request.getArtifact();
+        String version = artifact.getVersion();
+        if (!result.isAvailable() && SrcVersion.isSrcVersion(version)) {
+            Configuration config = configuration.getConfiguration();
+
+            ScmRepository scmRepo = configuration.findScmRepo(artifact);
+            Path projectBuildDir = scrdepsDir.resolve(scmRepo.getId());
+
+            BuilderIo builderIo = config.getBuilderIo();
+            IoRedirects ioRedirects = IoRedirects.builder() //
```

Comment:
I'll never understand why the "//" are required at the end of the line, to satisfy a linting tool :)